src/ci/scripts/install-msys2.sh

```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
-# Clean up and prepare the MSYS2 installation. MSYS2 is needed primarily for
-# the test suite (run-make), but is also used by the MinGW toolchain for assembling things.
+# Clean up and prepare the MSYS2 installation.
+# MSYS2 is used by the MinGW toolchain for assembling things.
```

Comment:
I don't see MSYS2 anywhere in this script apart from the comment and filename, probably could be renamed to  `setup-windows-python.sh` or something like that.

Reply:
Yeah, I'm not convinced this script is still needed at all but I didn't want to change too much at once. Or if it is just needed by the mingw targets then it could be merged into the other script.

Reply:
I think all Windows jobs need Python setup one way or another and after these changes this script is basically a custom `actions/setup-python` step: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python#specifying-a-python-version

Reply:
Hm, that says "To maintain consistent behavior with other runners and to allow Python to be used out-of-the-box without the setup-python action, GitHub adds a few versions from the tools cache to PATH"

Consistent behaviour with other runners sounds like something we can rely on, no?

Reply:
We use custom Docker images on Linux builders, no idea about Mac. So this consistency won't really apply here.

Reply:
Ah, I see. I think I'll experiment with maybe using the action in a follow up PR.